cmd/agent/app/reporter/tchannel/reporter.go

```diff
@@ -75,13 +84,23 @@ func NewTChannelReporter(svc string, channel *tchannel.Channel, mFactory metrics
 		metrics.Init(&bm, nsByType, nil)
 		batchesMetrics[s] = bm
 	}
-	rep := &tchannelReporter{zClient: zClient, jClient: jClient, logger: zlogger, batchesMetrics: batchesMetrics}
+	return &Reporter{
+		channel:        channel,
+		zClient:        zClient,
+		jClient:        jClient,
+		peerListMgr:    peerListMgr,
```

Comment:
it's not, just like "discovery client" wasn't used by the Agent. But it needs to be stored somewhere, and it logically belongs to the reporter since it (or rather the builder) creates it and uses it to manage its channel.

Also, in theory there could be a Close() function in the reporter that would call Close on peerListMgr so that it can unsubscribe from the discovery notifier. But this is an improvement outside of the scope of this PR.